nevergrad/optimization/test_optimizerlib.py

```diff
@@ -197,6 +197,86 @@ def test_infnan(name: str) -> None:
         result < 2.0, f"{name} failed and got {result} with {recom.value} (type is {type(optim)})."
 
 
+def suggestable(name: str) -> bool:
+    # Some methods are not good with suggestions.
+    if "TBPSA" in name:
+        return False
+    if "EMNA" in name or "EDA" in name:
+        return False
+    if "Stupid" in name:
+        return False
+    if "BO" in name:
+        return False
+    if "Pymoo" in name:
+        return False
+    return True
+
+
+@skip_win_perf  # type: ignore
+@pytest.mark.parametrize("name", [r for r in registry if suggestable(r)])  # type: ignore
+def test_suggest_optimizers(name: str) -> None:
+    """Checks that each optimizer is able to converge when optimum is given"""
+
+    optimizer_cls = registry[name]
+    instrum = ng.p.Array(shape=(100,)).set_bounds(0.0, 1.0)
+    instrum.set_integer_casting()
+    xs = np.asarray([0] * 17 + [1] * 17 + [0] * 66)
+    optim = optimizer_cls(instrum, budget=7)
+    target = lambda x: 0 if np.all(np.asarray(x, dtype=int) == xs) else 1
+    optim.suggest(xs)
+    optim.minimize(target)
+    assert np.all(optim.recommend().value == xs), "{name} proposes {optim.recommend().value} instead of {xs}"
```

Comment:
```suggestion
    assert not target(optim.recommend().value), "{name} proposes {optim.recommend().value} instead of {xs}"
```

Reply:
I don't get it, your version does not have xs anywhere in the equation ?

Reply:
Oh I see, you want to use the fact that the optimum has value 0.